nautobot/ipam/views.py

```diff
@@ -236,84 +239,63 @@ def get_extra_context(self, request, instance):
-    table = tables.VRFTable
-
-
-class VRFView(generic.ObjectView):
-    queryset = VRF.objects.all()
-
-    def get_extra_context(self, request, instance):
-        context = super().get_extra_context(request, instance)
-
-        prefixes = instance.prefixes.restrict(request.user, "view")
-        prefix_count = prefixes.count()
-        prefix_table = tables.PrefixTable(prefixes.select_related("namespace"), hide_hierarchy_ui=True)
-
-        # devices = instance.devices.restrict(request.user, "view")
-        # device_count = devices.count()
-        # device_table = DeviceTable(devices.all(), orderable=False)
-
-        import_targets_table = tables.RouteTargetTable(
-            instance.import_targets.select_related("tenant"), orderable=False
-        )
-        export_targets_table = tables.RouteTargetTable(
-            instance.export_targets.select_related("tenant"), orderable=False
-        )
-
-        # TODO(jathan): This table might need to live on Device and on VRFs
-        # (possibly replacing `device_table` above.
-        vrfs = instance.device_assignments.restrict(request.user, "view")
-        vrf_table = tables.VRFDeviceAssignmentTable(vrfs)
-        vrf_table.exclude = ("vrf",)
-        # context["vrf_table"] = vrf_table
-
-        paginate = {
-            "paginator_class": EnhancedPaginator,
-            "per_page": get_paginate_count(request),
-        }
-        RequestConfig(request, paginate).configure(prefix_table)
-        RequestConfig(request, paginate).configure(vrf_table)
-        RequestConfig(request, paginate).configure(import_targets_table)
-        RequestConfig(request, paginate).configure(export_targets_table)
-
-        context.update(
-            {
-                "device_table": vrf_table,
-                # "device_table": device_table,
-                "prefix_count": prefix_count,
-                "prefix_table": prefix_table,
-                "import_targets_table": import_targets_table,
-                "export_targets_table": export_targets_table,
-            }
-        )
-
-        return context
-
-
-class VRFEditView(generic.ObjectEditView):
-    queryset = VRF.objects.all()
-    model_form = forms.VRFForm
-    template_name = "ipam/vrf_edit.html"
-
-
-class VRFDeleteView(generic.ObjectDeleteView):
-    queryset = VRF.objects.all()
-
-
-class VRFBulkImportView(generic.BulkImportView):  # 3.0 TODO: remove, unused
-    queryset = VRF.objects.all()
-    table = tables.VRFTable
-
-
-class VRFBulkEditView(generic.BulkEditView):
-    queryset = VRF.objects.select_related("tenant")
-    filterset = filters.VRFFilterSet
-    table = tables.VRFTable
-    form = forms.VRFBulkEditForm
+    filterset_class = filters.VRFFilterSet
+    filterset_form_class = forms.VRFFilterForm
+    table_class = tables.VRFTable
+    form_class = forms.VRFForm
+    bulk_update_form_class = forms.VRFBulkEditForm
+    serializer_class = serializers.VRFSerializer
+
+    object_detail_content = ObjectDetailContent(
```

Comment:
Does it make sense to keep the same pattern of importing these classes from another file, like the other 7 attributes on the NautobotUIViewSet are?

Reply:
Yeah, that's fair.